### ISSUE TITLE
New office theme

### ISSUE THREAD

**Issue by korneliuscode** [fluentribbon/Fluent.Ribbon#957]
Hi, Are there any plans to support the new office theme?

Here is more detail on it: https://insider.office.com/en-us/blog/visual-refresh-of-office-apps-for-windows

**Comment by batzen** [comment]
Hi,
the simplified styles are currently develop in #924.
After that is done i will have a look at the changes Microsoft made and decide which changes in will adopt for Fluent.Ribbon.
The community of Fluent.Ribbon will then have to decide which path is taken.
As i won't, ever again, maintain completely different themes (rounded corners etc.).
I did that for years with the Office 2010 and 2013 themes and it was way too much work.
So the decision will have to be made if everything is changed to the new styles or if we find a good looking middle ground between the current theme and the new theme.

**Comment by nlogozzo** [comment]
@batzen I think it's time to address this.
With Windows 11 here with its new styles and the new office refresh in effect, I feel like moving Fluent.Ribbon to this new rounded and visual style will greatly modernize our WPF apps and allow us to keep in touch with Windows 11's design.

**Comment by nlogozzo** [comment]
And if not fully this:
![image](https://user-images.githubusercontent.com/17648453/168453428-5861d684-e0c7-4c2d-a258-98c3719f0bab.png)

Then maybe something like this for a visual refresh of this library:
![image](https://user-images.githubusercontent.com/17648453/168453422-6e51d3cf-1e79-4e4a-94ad-322db94362fd.png)

**Comment by jonasnordlund** [comment]
I understand if this is a lot of work but I don't think this new Windows 11 style is going away anytime soon and this would make Fluent.Ribbon a great pairing with other "modern WPF" projects like [WPFUI](https://github.com/lepoco/wpfui).

**Comment by nlogozzo** [comment]
I think you hit the nail on the head with what should change.
Just a little refresh similar to this:
![image](https://user-images.githubusercontent.com/17648453/171445460-59a60f3d-f974-4c88-9998-2f26d883e146.png)
Especially, `Colored "underscore" on selected tabs instead of connected background color` and just more roundness in general

**Comment by jonasnordlund** [comment]
Yes, no need for overwhelming changes, really. Font face should give priority to Segoe UI Variable, I think? Potentially animated ribbon focus indicator changes (I don't remember if "New Office" has this but Windows 11 in general does). Title bar and ribbon group padding could perhaps be configurable to cater for both compact and Microsoft style preferences (Windows 11 in general tend to have greater title bar paddings, like its Settings app). In fact, WinUI has a [compact mode flag](https://docs.microsoft.com/en-us/windows/apps/design/style/spacing#fluent-compact-sizing) that could perhaps be a borrowed idea.

**Comment by kevwkev** [comment]
![image](https://user-images.githubusercontent.com/1664903/174492728-6672d151-a36e-4276-839b-350fb9eb2fe5.png)

FYI: The "Ultimate UI for WPF" components from Infragistics also has a slick looking ribbon.

**Comment by batzen** [comment]
Just as an update.
I tried a few things and the current result doesn't look that bad.
What do you guys think?

![image](https://github.com/fluentribbon/Fluent.Ribbon/assets/1971030/95b4d993-3761-4660-bc39-126fe85460e5)

Using Segoe UI Variable or using Variable fonts in general is not possible in WPF as of now.
But support for it is listed in https://github.com/dotnet/wpf/issues/7813

**Comment by nlogozzo** [comment]
Looks great!
Maybe if the separators between groups (Clipboard, Font, etc...) can be more "defined" as they are here:
![image](https://github.com/fluentribbon/Fluent.Ribbon/assets/17648453/b9e17c04-8cea-4c13-9235-3a546d498dd3)

**Comment by kevwkev** [comment]
> Just as an update.
> 
> I tried a few things and the current result doesn't look that bad.
> 
> What do you guys think?
> 
> 
> 
> ![image](https://github.com/fluentribbon/Fluent.Ribbon/assets/1971030/95b4d993-3761-4660-bc39-126fe85460e5)
> 
> 
> 
> Using Segoe UI Variable or using Variable fonts in general is not possible in WPF as of now.
> 
> But support for it is listed in https://github.com/dotnet/wpf/issues/7813

This looks impressive already! Can't wait to see it in Simplified mode as well. To further enhance it, I have a few suggestions. As @nlogozzo mentioned, it would be great if the separator could be more defined or offering an option to adjust its color. This would add clarity and customization to the visual experience. Secondly, matching the width of the underline with the tab item header text will create a cohesive and balanced look. Lastly, increasing the margin for the buttons or providing an option to adjust the button's margin would improve spacing and readability. With these refinements, I believe we can achieve a sleek and modern style that truly stands out!

**Comment by batzen** [comment]
Will start working on this after my vacation.

**Comment by nlogozzo** [comment]
>  after my vacation.

Enjoy :)

**Comment by kevwkev** [comment]
Hey @batzen, I'm curious if there have been any updates regarding the new office theme. Could you consider sharing it on a new branch so that we can give it a test run?

**Comment by batzen** [comment]
I have a branch for that, but forgot to push it.
Will do that during the weekend.
If it's still not there after the weekend feel free to remind me again 😉

**Comment by batzen** [comment]
The changes are now in the develop branch.
Would be nice if you could provide feedback.
@kevwkev @nlogozzo @jonasnordlund @korneliuscode @GeertvanHorrik @robertmuehsig

**Comment by korneliuscode** [comment]
Thanks @batzen , i looked at it and think its looking pretty neat!

**Comment by batzen** [comment]
@korneliuscode Mica effect will follow next. 😁

**Comment by korneliuscode** [comment]
Oh, that would be great!!

**Comment by batzen** [comment]
@korneliuscode I added Mica last week. Would love to hear your feedback.

**Comment by korneliuscode** [comment]
I like it a lot. Thanks for adding!!

**Comment by nlogozzo** [comment]
I'm not able to test this right now... could we get some screenshots of what it looks like?

**Comment by batzen** [comment]
You can always grab the latest CI build, including the showcase application, from https://ci.appveyor.com/project/batzen/fluent-ribbon/branch/develop/artifacts

**Comment by d-griet** [comment]
> You can always grab the latest CI build, including the showcase application, from https://ci.appveyor.com/project/batzen/fluent-ribbon/branch/develop/artifacts

I've been using the latest `develop` branch for some time now, it seems to work without issues. Really like the new office theme and the Mica theming options. Are you planning on creating a new release anytime soon?

**Comment by batzen** [comment]
@d-griet Will release a preview version to nuget later today. Not sure about the final release. But i guess, depending on the feedback on the preview version, in about 2 months.

**Comment by kevwkev** [comment]
@batzen Thank you for refreshing the UI; it looks fantastic. However, I believe there are some improvements that could make it more closely resemble Microsoft's official implementation. I came across an open-source library, Coho.UI https://github.com/sebbouez/Coho.UI, which has a very polished appearance of the ribbon control. Could you take a look at it and consider borrowing some ideas? Specifically, its handling of mouse-over highlights with rounded corners, button spacing, and the transparency of the background when a tab is selected. Honestly, it is the only open-source UI library I've found that closely replicates the official Microsoft version. Thank you again for all your hard work! Below are some screenshots of Coho.UI’s ribbon for reference.
<img width="988" alt="Screenshot 2024-06-23 at 9 54 23 AM" src="https://github.com/fluentribbon/Fluent.Ribbon/assets/1664903/95b770b7-e280-40df-b4a6-113093c62313">
<img width="987" alt="Screenshot 2024-06-23 at 9 55 45 AM" src="https://github.com/fluentribbon/Fluent.Ribbon/assets/1664903/3f5771a7-815f-4340-88a2-db4bf958788f">

**Comment by qian-o** [comment]
When the new theme is floating, the panel will have a filled background. This should be a bug.

![1d5c1f35-5556-4fe0-bb7b-127e029e14f7](https://github.com/user-attachments/assets/c49986dd-5783-4f29-a584-6e440d38de6c)

**Comment by jonasnordlund** [comment]
I snuck in the RC with this in a beta of our enterprise application, which saw some use during one of our courses for new users. We've also trialled it internally for a few weeks now. So far, so good! Managed to get auto-dark/light mode detection going too (like when you change it live in the Windows 11 Settings), and it's also working well and synchronized with our AvalonDock component with very little work!

**Comment by batzen** [comment]
@qian-o The filled background is fixed in the latest preview.
